### ISSUE TITLE
Crash while walking around.

### ISSUE THREAD

**Issue by drayath** [CleverRaven/Cataclysm-DDA#29515]
**Describe the bug**  
Walking around on the map game crashed, have had it happen several times in completely different areas of the map.

Version: cataclysmdda-0.D-8774 (Windows 10-64 Tiles)  [Also occurred in 8712]
Mods:   "dda",  "aftershock",  "filthy_morale",  "no_npc_food",  "growable-pots",  "makeshift",  "Medieval_Stuff",  "More_Survival_Tools",  "modular_turrets",  "Salvaged_Robots",  "mutant_npcs",  "more_locations",  "FujiStruct",  "Urban_Development",  "deoxymod",  "blazemod",  "Tanks",  "no_flaming_weapons"
Options: Z-Levels(On), NPC static & dynamic (On)

![crash](https://user-images.githubusercontent.com/6987850/56076635-86092000-5dcb-11e9-832d-4dfe1b5d7da4.jpg)
Which is not logged as it is immediately followed by this one.

```
CRASH LOG FILE: config/crash.log
VERSION: 0.D-1723-g0c2f4b5
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:
    @0x4E59B0[cataclysm-tiles.exe+0xE59B0]
    @0x4E5D51[cataclysm-tiles.exe+0xE5D51]
    SMPEG_error+0x3DAF3@0xB54A95[cataclysm-tiles.exe+0x754A95]
    UnhandledExceptionFilter+0x1A0@0x77155F10[KERNELBASE.dll+0x195F10]
    RtlCaptureStackContext+0x1E521@0x777DDA71[ntdll.dll+0x9DA71]
    RtlAreBitsSet+0x58@0x777A4198[ntdll.dll+0x64198]
```

**Comment by drayath** [comment]
On a separate not is it possible for the build server to produce a windows symbol file for each build. So we can provide you with some better crash info.
Or ever better if more work automatically decode symbols in the crash log if file is available locally, e.g https://docs.microsoft.com/en-us/windows/desktop/debug/retrieving-symbol-information-by-address)

**Comment by ifreund** [comment]
Looks like a possible duplicate of #29049

**Comment by neitsa** [comment]
Damn, too bad, the stack is off by multiple order of magnitude, I couldn't get something out of it. :(

As for the symbols, the build chain is Linux only (using cross-compilation on linux to produce Windows executables), and none the tools can produce symbolic information for Windows. I have a made a detailed issue here: #29248

I'm expecting to be able to give it a try next week.

**Comment by neitsa** [comment]
Hmm ok, decided to give a try to the **x86** version and it matches the stack trace... 

The problem is that the double crash prevents a clear reading of the problem as we don't have the exact crash location 😢 .

* 0x598bdd: `game::update_map(int &,int &)` calls `map::shift(int,int)`
* 0x720494: `map::shift(int,int)` calls `map::loadn(int,int,int,bool)`
* 0x71FB9E: `map::loadn(int,int,int,bool)` calls `tinymap::tinymap(int,bool)`
* 0x71d526: `tinymap::tinymap(int, bool)` calls `map::map(int,bool)`
* exception somewhere in `map::map()` ?
* 0xd8f5e0: C runtime `__cxxabiv1::__terminate(void (*)(void))` calls `abort()`

As for the second stack trace it's already somewhere in the exception handler, so the exception has already been triggered at that point.

**Comment by drayath** [comment]
Eventually managed to get it to happen again, here is a minidump (from when the first crash dialog is up on screen)

Minidump: [cataclysm-tiles.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3076310/cataclysm-tiles.zip)
This should be the .core file for the first error: [core.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3076315/core.zip)

Compiled a windows-debug build and failed get it to occur there, mostly because moving each square takes a few seconds limiting exploration. (50-90% of cputime in the player_map_memory.memorize_tile() function though this might just be a debug-build artifact).
Couple of likely irrelevant observations: map::map() is always called with zLevels=false even though zLevels are enabled.
mapmemory.tilecache is holding around 4 Million items (Bionic Memory)

**Comment by kevingranade** [comment]
Did you have auto-forage on? I fixed a crash related to it in #29753 related to the auto-forage code, if you had it enabled I strongly suspect your issue is fixed as well.

**Comment by Night-Pryanik** [comment]
No confirmation in half a year, closing.
